### PR TITLE
Subscription form

### PR DESCRIPTION
**Pull request by amritsiidhu**
Subscription form on homepage
Tests for form

### CODE REVIEW

**Comment by amritsiidhu** [comment]
Test: When submitting the form it hits the api
Problem: The form submits to the subscription page where there is another button to submit to the api. I have not implemented the code to submit to the api. in the code behind the view. And i am not sure how to mock that up within the test. The commented out test currently submits the email address to the subscription page but i am not sure what to do from there. I tried mimicking your other tests to mock the api but have got lost.

**Comment by kenlt-uk** [comment]
You need to implement the submission to the api in order to test it, no point in mocking it. 
The call to the api can then be mocked once it is implemented.

You have already proven in other tests that you can reach the subscription page with the correct email passed through so no point in replicating it. Your test should just be that on the subscription page with the given email once the user submits it should hit the api call with the correct parameters and then probably get redirected back to the home page.

**Comment by amritsiidhu** [comment]
So in the code behind the subscription view, how do i there make it know to submit to api? I assume i need something like `if form.validate_on_submit()` post to api. But not sure how to do that

I have created methods in `api_client.py` called `get_subscription_email` `add_subscription_email`.

**Comment by kentsanggds** [comment]
When you do a validate on submit on the subscription page you need to make a call to the api_client method for submitting details for a new subscriber. Similar to this - https://github.com/NewAcropolis/frontend/blob/master/app/main/views/admin.py#L74-L99

**Comment by amritsiidhu** [comment]
The tests are all finally working, i can now successfully submit to the api and that goes back to the homepage, but how do i appropriately do the assert to show that. At the moment my assert just shows that it lands back on the homepage

**Comment by kenlt-uk** [comment]
One way to show that the api is being correctly called you need to check what arguments are being passed to the mock.

Also I have noticed that you have some merge conflict issues which will need to be cleared up.

**Comment by amritsiidhu** [comment]
I have rebased master. I have also added an assert that check the api is being correctly called by checking what arguments are being passed to the mock.

**Comment by kentsanggds** [comment]
Rather than - `Do you wish to subscribe to the Acropolis newsletter?` on the subscription page, perhaps update the text to `Click submit to subscribe to the newsletter:` and also remove the Text `Email: xxx@yyy` as I don't think its necessary.

**Comment by AmritSidhu** [comment]
I squashed the commits down. I added the form to the homepage in the magazine section. But when i try and run the tests i get `./scripts/run_tests.sh: line 17: flake8: command not found`. Any idea why this has happened?

**Comment by amritsiidhu** [comment]
Ive squashed all the commits. Got all the tests working. Added the form to the homepage in the magazine section as well. Made the changes to the subscription page you suggested. I think this is officially done now unless you can think of anything else?

File: app/clients/api_client.py [hunk 1]
```diff
@@ -109,3 +109,10 @@ def update_user_access_area(self, user_id, access_area):
             'access_area': access_area
         }
         return self.post(url='user/{}'.format(user_id), data=data)
+
+    def add_subscription_email(self, email):
+        data = {
+            'email': email
+        }
+        print(data)
```
Comment (kentsanggds) [inline app/clients/api_client.py]: This print can be removed

File: app/main/views/subscription.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+@main.route('/subscription', methods=['GET', 'POST'])
+def subscription():
+    subscription_form = SubscriptionForm()
+
+    if subscription_form.validate_on_submit():
+        try:
+            api_client.add_subscription_email(subscription_form.email.data)
+            return redirect(url_for('.index'))
+        except Exception as e:
+            print("EXCEPTION")
```
Comment (kentsanggds) [inline app/main/views/subscription.py]: prints should be taken out

File: app/templates/views/subscription.html [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{% extends "base.html" %}
+{% block title %}Subscription{% endblock %}
+
+{% block content %}
+
+<div class="container-fluid padding">
+  <div class="row padding text-center ">
+    <div class="col-xs-12 col-sm-12 col-md-6">
+       <h2> Click submit to subscribe to the newsletter:</h2><br>
```
Comment (kenlt-uk) [inline app/templates/views/subscription.html]: I've had a word with the team at New Acroopolis, and they have asked for a few changes to the wording:

Change `newsletter` be changed to `magazine` and `Submit` to `Subscribe`

And also can the whole phrase be changed to - 

`By clicking subscribe you will receive our magazine and information about our events:`

**Comment by AmritSidhu** [comment]
I have removed the debug prints, i have changed the the subscription page wording and i have rebased with master

**Pull request merged**